### PR TITLE
(GH-14) Improved naming of test cases. Missing tests added.

### LINKED ISSUE #14: Improve naming of test cases
Use a more natural language for naming test cases (e.g. `Action_Should_Not_Be_Called_If_Maybe_Value_Is_Not_Set` instead of `MaybeValueNotSet_ActionNotCalled`)

### PR DESCRIPTION
**Pull request by stefan-lindegger**
Improved naming of test cases. Missing tests added.
Fixes #14.

### CODE REVIEW

File: src/BBT.Maybe/NoneCase.cs [hunk 1]
```diff
@@ -35,7 +35,7 @@ internal NoneCase(bool isNone)
         /// <param name="b">Maybe to compare.</param>
         public static bool operator !=(NoneCase a, NoneCase b)
         {
-            return a != b;
+            return !(a == b);
```
Comment (pascalberger) [inline src/BBT.Maybe/NoneCase.cs:38]: Is this change intended?

File: src/BBT.Maybe/MaybeStruct.cs [hunk 2]
```diff
@@ -59,7 +60,7 @@ private MaybeStruct(SerializationInfo info, StreamingContext context)
         /// <param name="b">Maybe to compare.</param>
         public static bool operator !=(MaybeStruct<T> a, MaybeStruct<T> b)
         {
-            return a != b;
+            return !(a == b);
```
Comment (pascalberger) [inline src/BBT.Maybe/MaybeStruct.cs:63]: Is this change intended?

File: src/BBT.Maybe/Maybe.cs [hunk 3]
```diff
@@ -59,7 +60,7 @@ private Maybe(SerializationInfo info, StreamingContext context)
         /// <param name="b">Maybe to compare.</param>
         public static bool operator !=(Maybe<T> a, Maybe<T> b)
         {
-            return a != b;
+            return !(a == b);
```
Comment (pascalberger) [inline src/BBT.Maybe/Maybe.cs:63]: Is this change intended?

**Review by pascalberger — APPROVED** [review]
I've rebased on latest develop.

Some questions, otherwise LGTM

File: src/BBT.Maybe/NoneCase.cs (hunk 1 above)
Comment (stefan-lindegger) [reply src/BBT.Maybe/NoneCase.cs:38]: yes, recursive call of operator causes stack overflow exception.

File: src/BBT.Maybe/MaybeStruct.cs (hunk 2 above)
Comment (stefan-lindegger) [reply src/BBT.Maybe/MaybeStruct.cs:63]: yes, recursive call of operator causes stack overflow exception.

File: src/BBT.Maybe/Maybe.cs (hunk 3 above)
Comment (stefan-lindegger) [reply src/BBT.Maybe/Maybe.cs:63]: yes, recursive call of operator causes stack overflow exception.

**Pull request merged**